qemu/tests/mmu_basic.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test to disabling radix MMU mode on guest.
+    Steps:
+    1) There are two options, boot up a native(radix)guest or HPT guest.
+    2) Check the MMU mode in the guest.
+    3) Adding disable radix to guest's kernel line directly then reboot guest.
+    4) Check again the MMU mode in the guest.
+    5) Check guest call trace in dmesg log.
+
+
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    def _check_mmu():
+        mmu_mode = re.search(r'\w{4,}', session.cmd_output(params["mmu_check"]))
+        return mmu_mode.group(0) if mmu_mode else ""
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    if utils_test.check_kernel_cmdline(session, remove_args="disable_radix"):
+        utils_test.update_boot_option(vm, args_removed="disable_radix")
+        session = vm.wait_for_login()
+
+    error_context.context("Check the MMU mode.", logging.info)
+    pre_mmu_mode = _check_mmu()
+    if params["guest_type"] == "HPT":
+        if pre_mmu_mode != params["mmu"]:
+            test.fail("mmu mode is %s, doesn't meet expectations." % pre_mmu_mode)
+    else:
+        if pre_mmu_mode != params["pre_mmu"]:
+            test.fail("mmu mode is %s, doesn't meet expectations." % pre_mmu_mode)
+
+    utils_test.update_boot_option(vm, args_added="disable_radix")
+    session = vm.wait_for_login()
+    error_context.context("Check the MMU mode.", logging.info)
+    if _check_mmu() != params["mmu"]:
+        test.fail("mmu mode is %s, mmu mode disabled failure." % _check_mmu())
+
+    utils_test.update_boot_option(vm, args_removed="disable_radix")
```

Comment:
Should be `args_added`

Reply:
Here I want to restore the kernel parameter to a state where it is not added, so I think it should be args_removed, right?

Reply:
OK, you are right. But there has a issue, if this case failed before this step, `disable_radix` will still in guest kernel command line, so suggest use `image_snapshot`

Reply:
Thanks, I also thought about try: finally:,  image_snapshot is indeed safer for this case.
correct